Reduce InjectedBundle SPI surface area

achristensen07 opened the pull request:
#### 882f2782a7e91d861674f0c8d08fb6a3161db98d
<pre>
Reduce InjectedBundle SPI surface area
<a href="https://bugs.webkit.org/show_bug.cgi?id=298644">https://bugs.webkit.org/show_bug.cgi?id=298644</a>
<a href="https://rdar.apple.com/160260932">rdar://160260932</a>

Reviewed by Carlos Garcia Campos.

<a href="https://rdar.apple.com/160170207">rdar://160170207</a> removed the last uses of these SPI endpoints:

WKBundlePageUIClient.mouseDidMoveOverElement
WKBundleHitTestResultGetFrame
WKBundleHitTestResultGetTargetFrame
WKBundleHitTestResultCopyImage

We can now remove support for them in WebKit, along with their associated tests.

A few build fixes are necessary because of the change in unified builds,
including a surprising one in NativeImage.h and one in WKScrollGeometryTests.mm

* Source/WebCore/platform/graphics/NativeImage.h:
* Source/WebKit/UIProcess/API/APIUIClient.h:
(API::UIClient::mouseDidMoveOverElement):
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetPageUIClient):
* Source/WebKit/UIProcess/Cocoa/UIDelegate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::UIClient::mouseDidMoveOverElement):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::dispatchMouseDidMoveOverElementAsynchronously):
(WebKit::WebPageProxy::mouseDidMoveOverElement):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundlePageUIClient.h:
(API::InjectedBundle::PageUIClient::mouseDidMoveOverElement): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleHitTestResult.cpp:
(WKBundleHitTestResultGetFrame):
(WKBundleHitTestResultGetTargetFrame):
(WKBundleHitTestResultCopyImage):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleHitTestResult.h:
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePageUIClient.cpp:
(WebKit::InjectedBundlePageUIClient::mouseDidMoveOverElement): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePageUIClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::mouseDidMoveOverElement):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::performHitTestForMouseEvent):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FrameHandleSerialization.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
((WebKit, MouseMoveOverElement)):
((WebKit, MouseMoveOverElementWithClosedWebView)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKScrollGeometryTests.mm:

Canonical link: <a href="https://commits.webkit.org/299809@main">https://commits.webkit.org/299809@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eccedff2e0458e6f4fa271acbc2ad89a08a610c1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120217 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39911 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30562 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126572 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72290 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0309dd5e-bded-4bfb-9f98-8c93702bfa32) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122093 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40607 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48488 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91292 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60589 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ceeeb3fb-391b-4ac2-8542-cfda8175dd67) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123169 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32429 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107780 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71846 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d62e28ab-1fe6-4cf0-a5f0-2991f9569daf) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31463 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25886 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70197 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101909 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26071 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129471 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47137 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35761 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99916 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47503 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103965 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99758 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45209 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23235 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43768 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19105 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46999 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52705 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46467 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49814 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48151 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c3972f2d6fba2b8f358de31085e17848a534e95e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50530

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/eccedff2e0458e6f4fa271acbc2ad89a08a610c1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50530

achristensen07 committed:
Reduce InjectedBundle SPI surface area
https://bugs.webkit.org/show_bug.cgi?id=298644
rdar://160260932

Reviewed by Carlos Garcia Campos.

rdar://160170207 removed the last uses of these SPI endpoints:

WKBundlePageUIClient.mouseDidMoveOverElement
WKBundleHitTestResultGetFrame
WKBundleHitTestResultGetTargetFrame
WKBundleHitTestResultCopyImage

We can now remove support for them in WebKit, along with their associated tests.

A few build fixes are necessary because of the change in unified builds,
including a surprising one in NativeImage.h and one in WKScrollGeometryTests.mm

* Source/WebCore/platform/graphics/NativeImage.h:
* Source/WebKit/UIProcess/API/APIUIClient.h:
(API::UIClient::mouseDidMoveOverElement):
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetPageUIClient):
* Source/WebKit/UIProcess/Cocoa/UIDelegate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::UIClient::mouseDidMoveOverElement):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::dispatchMouseDidMoveOverElementAsynchronously):
(WebKit::WebPageProxy::mouseDidMoveOverElement):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundlePageUIClient.h:
(API::InjectedBundle::PageUIClient::mouseDidMoveOverElement): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleHitTestResult.cpp:
(WKBundleHitTestResultGetFrame):
(WKBundleHitTestResultGetTargetFrame):
(WKBundleHitTestResultCopyImage):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleHitTestResult.h:
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePageUIClient.cpp:
(WebKit::InjectedBundlePageUIClient::mouseDidMoveOverElement): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePageUIClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::mouseDidMoveOverElement):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::performHitTestForMouseEvent):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FrameHandleSerialization.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
((WebKit, MouseMoveOverElement)):
((WebKit, MouseMoveOverElementWithClosedWebView)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKScrollGeometryTests.mm:

Canonical link: https://commits.webkit.org/299809@main
